src/mscorlib/shared/System/StringSpanHelpers.cs

```diff
@@ -89,5 +89,28 @@ public static int IndexOf(this ReadOnlySpan<char> source, char value, int startI
 
             return -1;
         }
+
+        public static ReadOnlySpan<char> Remove(this ReadOnlySpan<char> source, int startIndex, int count)
+        {
+            if (startIndex < 0) throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_StartIndex);
+            if (count < 0) throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NegativeCount);
+            if (count > source.Length - startIndex) throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_IndexCount);
```

Comment:
I am seeing Remove is internal helper method which is called by us. do we need to do this validation every time it is called? can we just assert for these?

Reply:
> can we just assert for these?

We could.  I left those checks in because my hope is that it becomes public in https://github.com/dotnet/corefx/issues/21395, and so when adding these methods here, I basically just copied the existing string methods and then fixed them up to work with spans.